modules/phase_field/src/utils/MisorientationAngleCalculator.C

```diff
@@ -0,0 +1,212 @@
+        { 5.000e-01,  0.000e+00,  0.000e+00,  8.660e-01},
+        { 0.000e+00,  0.000e+00, -1.000e+00,  0.000e+00},
+        { 0.000e+00,  0.000e+00,  0.000e+00,  1.000e+00},
+        { 0.000e+00, -5.000e-01, -8.660e-01,  0.000e+00},
+        {-5.000e-01,  0.000e+00,  0.000e+00,  8.660e-01},
+        { 0.000e+00, -8.660e-01, -5.000e-01,  0.000e+00},
+        {-8.660e-01,  0.000e+00,  0.000e+00,  5.000e-01},
+        { 0.000e+00, -1.000e+00,  0.000e+00,  0.000e+00}
+      }; // 12 symmetric for hcp
+    else if ( quat_type == QuatType::getCSymm && crystal_type == CrystalType::FCC)
+      q_num = {
+        { 1.000E+00,	 0.000E+00, 	 0.000E+00, 	 0.000E+00},
+        { 5.000E-01,	 5.000E-01, 	 5.000E-01, 	 5.000E-01},
+        {-5.000E-01,	 5.000E-01, 	 5.000E-01, 	 5.000E-01},
+        { 6.123E-17,	 7.071E-01, 	 7.071E-01, 	 8.660E-17},
+        {-7.071E-01,	 1.543E-16, 	 7.071E-01, 	 9.881E-17},
+        {-7.071E-01,	-7.071E-01, 	 2.343E-16, 	 1.221E-17},
+        { 7.071E-01,	 0.000E+00, 	 0.000E+00, 	 7.071E-01},
+        { 1.110E-16,	 7.071E-01, 	 3.925E-17, 	 7.071E-01},
+        {-7.071E-01,	 7.071E-01, 	 5.551E-17, 	 2.680E-16},
+        {-1.793E-17,	 1.000E+00, 	 8.865E-17, 	 1.045E-16},
+        {-5.000E-01,	 5.000E-01, 	 5.000E-01, 	-5.000E-01},
+        {-5.000E-01,	-5.000E-01, 	 5.000E-01, 	-5.000E-01},
+        { 6.123E-17,	 0.000E+00, 	 0.000E+00, 	 1.000E+00},
+        {-5.000E-01,	 5.000E-01, 	-5.000E-01, 	 5.000E-01},
+        {-5.000E-01,	 5.000E-01, 	-5.000E-01, 	-5.000E-01},
+        {-8.660E-17,	 7.071E-01, 	-7.071E-01, 	 6.123E-17},
+        {-1.421E-16,	 7.071E-01, 	-1.110E-16, 	-7.071E-01},
+        {-5.551E-17,	 1.910E-16, 	 7.071E-01, 	-7.071E-01},
+        {-7.071E-01,	 0.000E+00, 	 0.000E+00, 	 7.071E-01},
+        {-7.071E-01,	 7.177E-17, 	-7.071E-01, 	 1.340E-16},
+        {-3.005E-16,	 5.551E-17, 	-7.071E-01, 	-7.071E-01},
+        {-1.045E-16,	 1.009E-16, 	-1.000E+00, 	-1.793E-17},
+        { 5.000E-01,	 5.000E-01, 	-5.000E-01, 	-5.000E-01},
+        { 5.000E-01,	 5.000E-01, 	 5.000E-01, 	-5.000E-01}
+      }; // 24 symmetric for fcc
+
+    std::vector<QuatReal> q(q_num.size());
+    for (const auto i : index_range(q_num))
+    {
+      q[i].w() = q_num[i][0];
+      q[i].x() = q_num[i][1];
+      q[i].y() = q_num[i][2];
+      q[i].z() = q_num[i][3];
+    }
+    return q;
+  }
+
+  Real dotQuaternion(const QuatReal & o1, const QuatReal & o2, 
+                                              const std::vector<QuatReal> & qcs, 
+                                              const std::vector<QuatReal> & qss)
+  {
+    Real d = 0.0; // used to get misorientation angle
+    QuatReal mori = itimesQuaternion(o1, o2);
+
+    if (qss.size() <= 1)
+      d = dotOuterQuaternion(mori, qcs); 
+    else
+      d = mtimes2Quaternion(o1, qcs, o2); // mtimesQuaternion(qss,mtimesQuaternion(o1,qcs,0),1)
+
+    return d;
+  }
+
+  QuatReal itimesQuaternion(const QuatReal & q1, const QuatReal & q2)
```

Comment:
sure that is doesn't already exist in libEigen?

Reply:
Yes, as I said before, I didn't make full use of the existing c++ library (libEigen), and I will optimize this part later.